PrestaShop/PrestaShop
# Simplified category form

| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Simplifies category form and refactors category saving, by adding more abstract classes. This makes sure that there is less repeated code between root and normal category management. Root category form type was already abstracted so there is little reason why form handlers shouldn't be abstracted as well.
| Type?         | refacto
| Category?     | BO
| BC breaks?    | yes
| Deprecations? | no
| Fixed ticket? | Part of #16482
| How to test?  | See below

## How to test

Simplifying Sell -> Catalog -> Category. Everything must look the same.

Please test the "add menu thumbnail images" capacity: add multiple, remove, and there is a limit (3) above which **the input is disabled**.

The description should have clean HTML(no javascript). Some fields can't exceed a certain number of characters meta title and meta keywords: 255, meta description: 512, name and link rewrite: 128 characters

## :notebook: BC Break
Backwards compatibility break introduced because AbstractCategoryType has changed parameters. This means that if any module extends this type(or any type that extends it) that module might crash upon updating to PS version where this update is present.

Backward comptability break introduced because RootCategoryFormDataHandler createAddCategoryCommand and createEditCategoryCommand functions are now protected. They are now protected because they were previously protected in CategoryFormDataHandler

---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/prestashop/prestashop/21243)
